mne/time_frequency/tfr.py

```diff
@@ -227,6 +227,237 @@ def _cwt(X, Ws, mode="same", decim=1, use_fft=True):
+    n_jobs : int, Defaults to 1
+        The number of epochs to process at the same time. The parallization is
+        implemented across channels.
+    verbose : bool, str, int, or None, Defaults to 'INFO'
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    out : array
+        Time frequency transform of epoch_data. If output is in ['complex',
+        'phase', 'power'], then shape of out is (n_epochs, n_chans, n_freqs,
+        n_times), else it is (n_chans, n_freqs, n_times). If output is
+        'avg_power_phaselock', the real values code for 'avg_power' and the
+        imaginary values code for the 'avg_phaselock':
+        out = avg_power + i * avg_phaselock
+    """
+    # Check data
+    epoch_data = np.asarray(epoch_data)
+    if epoch_data.ndim != 3:
+        raise ValueError('epoch_data must be of shape '
+                         '(n_epochs, n_chans, n_times)')
+
+    # Check frequencies
+    if not isinstance(frequencies, (list, np.ndarray)):
+        raise ValueError('frequencies must be an array-like.')
+    frequencies = np.asarray(frequencies, dtype=float)
+    if frequencies.ndim != 1:
+        raise ValueError('frequencies must be of shape n_freqs.')
+
+    # Check sfreq
+    if not isinstance(sfreq, (float, int)):
+        raise ValueError('sfreq must be a float or an int.')
+    sfreq = float(sfreq)
+
+    # Check params
+    allowed_ouput = ('complex', 'power', 'phase',
+                     'avg_power_phaselock', 'avg_power', 'avg_phaselock')
+    if output not in allowed_ouput:
+        raise ValueError("Unknown output type.")
+
+    if method not in ('mtm', 'morlet'):
+        raise ValueError('method must be "morlet" or "mtm"')
+
+    # Default zero_mean = True if mtm else False
+    zero_mean = method == 'mtm' if zero_mean is None else zero_mean
+    if not isinstance(zero_mean, bool):
+        raise ValueError('')
+    frequencies = np.asarray(frequencies)
+
+    # XXX Check not sure whether it makes sense to compute single-trial phases
+    # with mtm.
+    if (method == 'mtm') and (output == 'phase'):
+        raise NotImplementedError(
+            'This function is not optimized to compute the phase using the '
+            'mtm method. Use np.angle of the complex output instead.')
+
+    # Setup wavelet
+    if method == 'morlet':
+        W = morlet(sfreq, frequencies, n_cycles=n_cycles, zero_mean=zero_mean)
+        Ws = [W]  # to have same dimensionality as the 'mtm' case
+        if time_bandwidth != 4.0:
+            raise ValueError('time_bandwidth only applies to "mtm" method.')
+    elif method == 'mtm':
+        Ws = _dpss_wavelet(sfreq, frequencies, n_cycles=n_cycles,
+                           time_bandwidth=time_bandwidth, zero_mean=zero_mean)
+
+    # Check wavelets
+    if len(Ws[0][0]) > epoch_data.shape[2]:
+        raise ValueError('Wavelet is too long for such a short signal.')
```

Comment:
I was recently thinking it might be better phrased as "At least one of the wavelets is longer than the signal. Use longer signal or shorter wavelets."